.github/workflows/build_and_test.yaml

```diff
@@ -0,0 +1,202 @@
+name: build_and_test
+
+on:
+  workflow_call:
+    inputs:
+      build_type:
+        required: true
+        type: string
+
+permissions:
+  actions: read
+  checks: none
+  contents: read
+  deployments: none
+  discussions: none
+  id-token: write
+  issues: none
+  packages: read
+  pages: none
+  pull-requests: read
+  repository-projects: none
+  security-events: none
+  statuses: none
+
+jobs:
+  compute-build-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      MATRIX: ${{ steps.compute-matrix.outputs.MATRIX }}
+    steps:
+      - name: Compute Build Matrix
+        id: compute-matrix
+        run: |
+          set -eo pipefail
+
+          export MATRIX="
+          # amd64
+          - { ARCH: 'amd64', PY_VER: '3.11', CUDA_VER: '11.8.0', LINUX_VER: 'rockylinux8' }
+          - { ARCH: 'amd64', PY_VER: '3.11', CUDA_VER: '12.2.2', LINUX_VER: 'rockylinux8' }
+          # arm64
+          - { ARCH: 'arm64', PY_VER: '3.11', CUDA_VER: '11.8.0', LINUX_VER: 'rockylinux8' }
+          - { ARCH: 'arm64', PY_VER: '3.11', CUDA_VER: '12.2.2', LINUX_VER: 'rockylinux8' }
+          "
+
+          MATRIX="$(
+            yq -n -o json 'env(MATRIX)' | \
+            jq -c '{include: .}'
+          )"
+
+          echo "MATRIX=${MATRIX}" | tee --append "${GITHUB_OUTPUT}"
+  build:
+    name:  build-${{ matrix.CUDA_VER }}, ${{ matrix.ARCH }}, ${{ matrix.LINUX_VER }}
+    needs: compute-build-matrix
+    strategy:
+      matrix: ${{ fromJSON(needs.compute-build-matrix.outputs.MATRIX) }}
+    runs-on: "linux-${{ matrix.ARCH }}-cpu16"
+    container:
+      image: "rapidsai/ci-wheel:cuda${{ matrix.CUDA_VER }}-${{ matrix.LINUX_VER }}-py${{ matrix.PY_VER }}"
+      env:
+        RAPIDS_BUILD_TYPE: ${{ inputs.build_type }}
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.AWS_ROLE_ARN }}
+          aws-region: ${{ vars.AWS_REGION }}
+          role-duration-seconds: 43200 # 12h
+      - name: checkout code repo
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Get current date
+        id: date
+        run: |
+          echo "CURRENT_DATE=$(date --rfc-3339=date)" >> ${GITHUB_ENV}
+      - name: Standardize repository information
+        uses: rapidsai/shared-actions/rapids-github-info@main
+        with:
+          repo: ${{ github.repository }}
+          branch: ${{ github.ref_name }}
+          date: ${{ steps.date.outputs.date }}
+          sha: ${{ github.ref_name }}
+      - name: Build and repair the wheel
+        run: ci/build_wheel.sh
+        env:
+          GH_TOKEN: ${{ github.token }}
+          RAPIDS_BUILD_TYPE: ${{ inputs.build_type }}
+        # Use a shell that loads the rc file so that we get the compiler settings
+        shell: bash -leo pipefail {0}
+  compute-test-matrix:
```

Comment:
You could consider consolidating all the matrices into a single `compute-matrices` job.

I'm fine with either approach, just a thought.

Reply:
Combined, I like that idea.